aws/table_aws_route53_record.go

```diff
@@ -152,26 +152,36 @@ func listRoute53Records(ctx context.Context, d *plugin.QueryData, _ *plugin.Hydr
 		return nil, nil
 	}
 
+	maxItems := int32(100)
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 1 {
+				maxItems = int32(1)
+			} else {
+				maxItems = int32(limit)
+			}
+		}
```

Comment:
```suggestion
		if limit < maxItems {
			maxItems = int32(limit)
		}
```

I don't think we need to check if the limit is less than 1 anymore, as I ran some tests earlier and it seems like if someone runs with a query with `limit 0`, the hydrate functions aren't called.